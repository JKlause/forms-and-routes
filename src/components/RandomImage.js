import React from 'react';
// import styles from './Footer.css';
import PropTypes from 'prop-types';

const RandomImage = ({ photoUrl, name, handleRandomGet }) => {
  return (
    <>
      <img src={photoUrl} />
      <figcaption>{name}</figcaption>
      <button onClick={handleRandomGet}>Get New Random Character</button>
    </>
  );
};

RandomImage.propTypes = {
  photoUrl: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  handleRandomGet: PropTypes.func.isRequired
};


export default RandomImage;
