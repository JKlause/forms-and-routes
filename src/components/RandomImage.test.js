import React from 'react';
import { shallow } from 'enzyme';
import RandomImage from './RandomImage';

describe('RandomImage component', () => {
  const props = {
    photoUrl: '',
    name: '',
    handleRandomGet: () => {}
  };

  it('renders RandomImage', () => {   
    const wrapper = shallow(<RandomImage {...props}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
