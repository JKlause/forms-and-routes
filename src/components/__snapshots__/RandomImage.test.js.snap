// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RandomImage component renders RandomImage 1`] = `
<Fragment>
  <img
    src=""
  />
  <h2 />
  <button
    onClick={[Function]}
  >
    Get New Random Character
  </button>
</Fragment>
`;
